examples/platform/silabs/SilabsDeviceDataProvider.cpp

```diff
@@ -391,10 +391,7 @@ CHIP_ERROR SilabsDeviceDataProvider::GetHardwareVersionString(char * buf, size_t
 #if defined(CHIP_DEVICE_CONFIG_DEFAULT_DEVICE_HARDWARE_VERSION_STRING)
     if (err == CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND)
     {
-        VerifyOrReturnError(buf != nullptr, CHIP_ERROR_NO_MEMORY);
-        VerifyOrReturnError(bufSize > strlen(CHIP_DEVICE_CONFIG_DEFAULT_DEVICE_HARDWARE_VERSION_STRING),
-                            CHIP_ERROR_BUFFER_TOO_SMALL);
-        Platform::CopyString(buf, bufSize, CHIP_DEVICE_CONFIG_DEFAULT_DEVICE_HARDWARE_VERSION_STRING);
+        memcpy(buf, CHIP_DEVICE_CONFIG_DEFAULT_DEVICE_HARDWARE_VERSION_STRING, bufSize);
```

Comment:
Yeah @sharad-patil24 this file doesn't need any change. This change was only added to SLC since it was not present there. The change was already was done on CSA if I am not wrong.